ytypes/leaf_test.go

```diff
@@ -938,27 +938,29 @@ func TestValidateLeafRef(t *testing.T) {
 }
 
 type LeafContainerStruct struct {
-	Int8Leaf            *int8           `path:"int8-leaf"`
-	Int8LeafList        []int8          `path:"int8-leaflist"`
-	Uint8Leaf           *uint8          `path:"uint8-leaf"`
-	Int16Leaf           *int16          `path:"int16-leaf"`
-	Uint16Leaf          *uint16         `path:"uint16-leaf"`
-	Int32Leaf           *int32          `path:"int32-leaf"`
-	Uint32Leaf          *uint32         `path:"uint32-leaf"`
-	Int64Leaf           *int64          `path:"int64-leaf"`
-	Uint64Leaf          *uint64         `path:"uint64-leaf"`
-	StringLeaf          *string         `path:"string-leaf"`
-	BinaryLeaf          Binary          `path:"binary-leaf"`
-	BoolLeaf            *bool           `path:"bool-leaf"`
-	DecimalLeaf         *float64        `path:"decimal-leaf"`
-	EnumLeaf            EnumType        `path:"enum-leaf"`
-	UnionEnumLeaf       EnumType        `path:"union-enum-leaf"`
-	UnionLeaf           UnionLeafType   `path:"union-leaf"`
-	UnionLeaf2          *string         `path:"union-leaf2"`
-	EmptyLeaf           YANGEmpty       `path:"empty-leaf"`
-	UnionLeafSlice      []UnionLeafType `path:"union-leaflist"`
-	UnionLeafSingleType []string        `path:"union-stleaflist"`
-	UnionEnumLeaflist   []EnumType      `path:"union-enum-leaflist"`
+	Int8Leaf             *int8                 `path:"int8-leaf"`
+	Int8LeafList         []int8                `path:"int8-leaflist"`
+	Uint8Leaf            *uint8                `path:"uint8-leaf"`
+	Int16Leaf            *int16                `path:"int16-leaf"`
+	Uint16Leaf           *uint16               `path:"uint16-leaf"`
+	Int32Leaf            *int32                `path:"int32-leaf"`
+	Uint32Leaf           *uint32               `path:"uint32-leaf"`
+	Int64Leaf            *int64                `path:"int64-leaf"`
+	Uint64Leaf           *uint64               `path:"uint64-leaf"`
+	StringLeaf           *string               `path:"string-leaf"`
+	BinaryLeaf           Binary                `path:"binary-leaf"`
+	BoolLeaf             *bool                 `path:"bool-leaf"`
+	DecimalLeaf          *float64              `path:"decimal-leaf"`
+	EnumLeaf             EnumType              `path:"enum-leaf"`
+	UnionEnumLeaf        EnumType              `path:"union-enum-leaf"`
+	UnionLeaf            UnionLeafType         `path:"union-leaf"`
+	UnionLeaf2           *string               `path:"union-leaf2"`
+	EmptyLeaf            YANGEmpty             `path:"empty-leaf"`
+	UnionLeafSlice       []UnionLeafType       `path:"union-leaflist"`
+	UnionLeafSingleType  []string              `path:"union-stleaflist"`
```

Comment:
How does this type get generated now? In the previous approach we would generate this simply as `*string`, since there was no need to have a complex type here. Is there a change that has been made to the code that does that - such that we now create some basic type (or is this actually a `leaf-list` of string `union`)?

Reply:
The union name generation code is unchanged (see [last PR](https://github.com/openconfig/ygot/pull/451/files?file-filters%5B%5D=.go)). The union generation code is only changed for when `len(types) > 1`. So, the simple type situation is unchanged from before.

Reply:
Great, thanks.